pkg/operator/executor.go

```diff
@@ -6,7 +6,7 @@ import (
 )
 
 type phaser interface {
-	plan(ctx context.Context) error
+	plan(ctx context.Context) (alreadyApplied bool, err error)
```

Comment:
This is the most relevant part. The plan returns `alreadyApplied, err` instead of only `err`